### ISSUE TITLE
Segmentation fault due to Doctrine issue  

### ISSUE THREAD

**Issue by bibich** [api-platform/api-platform#1904]
**API Platform version(s) affected**: 2.6.4

**Description**  
Appear when persisting entity with a nullable ManyToOne relation

**How to reproduce**  
```php
<?php

namespace App\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use App\Repository\SkillRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=SkillRepository::class)
 */
#[ApiResource(
    collectionOperations: ["get"],
    itemOperations: ["get"],
    attributes: ["pagination_enabled" => false]
)]
class Skill
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $label;

    /**
     * @ORM\ManyToOne(targetEntity=Skill::class)
     * @ORM\JoinColumn(name="parent_id", referencedColumnName="id", nullable=true)
     */
    private $parent = null;
    
    // ...
}
```

```php
$skill = new Skill();
$skill->setLabel('Test');
$entityManager->persist($skill);
$entityManager->flush();
```
`\Doctrine\DBAL\Statement::bindValue` binds null value to integer instead of null 

**Possible Solution**  
?

**Additional Context**  
There is an opened issue on Doctrine repository : https://github.com/doctrine/orm/issues/8667

**Comment by bertrandjamin** [comment]
Hi @bibich  

I think that the problem comes from a wrong fix in php 8.0.5 which have created an issue on Doctrine.

https://bugs.php.net/bug.php?id=81002

The bug is already fixed but await for the release of php 8.0.6

If you have this problem with an other php version forget my comment.

**Comment by bibich** [comment]
Thanks @bertrandjamin  
You're right. 
Awaiting the release of 8.0.6, I switched back to php 8.0.3 and everything is fine.

`api/Dockerfile` :

```
ARG PHP_VERSION=8.0.3
...
```
